.github/workflows/docs-publish.yml

```diff
@@ -0,0 +1,57 @@
+# This workflow will publish API docs to GitHub Pages when a new release tag is added to the main branch
+
+name: Publish API docs to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    tags:
```

Comment:
we could change this so it only runs when a new tag is created (assuming we dont create tags that go back in time in semantic versioning)

```
on:
  push:
    branches:
      - main
  create:
    tags:
      - 'v*.*.*'
```

then you wouldn't need to do the custom check for the latest tag bash scripting

Reply:
We might create tags that go back in time in semantic versioning. For example, we release v2 and then need to release a patch to v1.